lib/ical/recur_iterator.js

```
@@ -179,7 +179,14 @@ class RecurIterator {
     this.initialized = options.initialized || false;
 
     if (!this.initialized) {
-      this.init();
+      try {
+        this.init();
+      } catch (e) {
+        // Init may error if there are no possible recurrence instances from the
+        // rule, but we don't want to bubble this error up. Instead, we create
+        // an empty iterator.
+        this.completed = true;
```

Would it make sense to use a custom error class instead and catch only that one? Otherwise it would hide other exceptions during development.

---

You're thinking using [this pattern](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch#conditional_catch_blocks)?

---

Yes